src/sec_security_common.c

```diff
@@ -1488,6 +1488,7 @@ Sec_Result SecDigest_SingleInput(Sec_ProcessorHandle *proc,
 	if (res != SEC_RESULT_SUCCESS)
 	{
 		SEC_LOG_ERROR("SecDigest_GetInstance failed");
+		SecDigest_Release(digest_handle, digest, digest_len);
 		return res;
 	}
```

Comment:
This should not be necessary if the SecDigest_GetInstance fails, nothing should have been allocated in that case.